main_test.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import "testing"
+
+func TestLongAdd(t *testing.T) {
+	for _, tc := range []struct {
+		name      string
+		a         string
+		b         string
+		sumResult string
+	}{
+
+		{"zero", "0", "0", "0"},
+		{"simple math", "123", "210", "333"},
+		{"Math with memorization ", "55555", "55", "55610"},
+		{"Nines", "999999999999999999999999999999999", "1", "1" + addZeros("", 33)},
```

Comment:
Please avoid using implementation functions for testing - it's possible to have a bug there, which would make it harder to detect. Unless you create a dedicated testsuite for that specific function, of course.